### PR TITLE
Fix synchronization when calling delegates

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] ~Has `CHANGELOG.md` been updated?~ (affects only wrapper SDKs)
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [ ] ~Did you check UI tests on the sample app? They are not executed on CI.~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

There is a deadlock between 
`MSChannelUnitDefault.enumerateDelegatesForSelector:selector withBlock:`  
(waits for `MSCrashes` and locks `delegates`)
and 
`MSCrashes.channel:didPrepareLog:internalId:flags:`
(locks `MSCrashes` and waits for `delegates`)

To reproduce:
```
for index in 0...30 {
      DispatchQueue.global().async {
          // track exception
   }
}
```

## Related PRs or issues

[AB#63536](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/63536)

### CODE REVIEW

File: AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m [hunk 1]
```diff
@@ -540,6 +541,62 @@ - (void)testLogsSentWithSuccess {
...
+
+  // If
+  NSObject *lock = [NSObject new], *syncCallback = [NSObject new], *syncBackground = [NSObject new];
+  [self initChannelEndJobExpectation];
+  id<MSLog> mockLog1 = [self getValidMockLog];
+  id<MSLog> mockLog2 = [self getValidMockLog];
+  id delegateMock = OCMProtocolMock(@protocol(MSChannelDelegate));
+  OCMStub([delegateMock channel:self.sut didPrepareLog:OCMOCK_ANY internalId:OCMOCK_ANY flags:MSFlagsDefault])
+      .andDo(^(__unused NSInvocation *invocation) {
+        // Notify that didPrepareLog has been called.
```
Comment (russelarms) [inline AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: Add a new line before comment.

Comment (clpolet) [reply AppCenter/AppCenterTests/MSChannelUnitDefaultTests.m]: It's the first line within a block so unfortunately our code formater won't accept the new line.

**Comment by MatkovIvan** [comment]
@annakocheshkova 

image


Let's update only wrapper changelogs

**Comment by guperrot** [comment]
> @annakocheshkova
> 
> image
> 
> Let's update only wrapper changelogs

Will be easy to forget, please do the changelog PR now on wrappers.

**Pull request merged**